client/jetpack-connect/signup.js

```diff
@@ -100,7 +100,7 @@ export class JetpackSignup extends Component {
 		debug( 'submitting new account', userData );
 		this.setState( { isCreatingAccount: true }, () =>
 			this.props
-				.createAccount( userData )
+				.createAccount( { ...userData, extra: { jpc: true } } )
```

Comment:
Hmm, seems like this way we'll erase what we already have in `userData.extra`, and it seems we already expect to have `first_name` and `last_name` there:

https://github.com/Automattic/wp-calypso/blob/9ab6a14fdfb6a3cf72b5ac4b339517aa34d57716/client/blocks/signup-form/index.jsx#L420-L423

Perhaps we could avoid that by doing something like:

```
.createAccount( { ...userData, extra: { ...userData.extra, jpc: true } } )
```

Reply:
Updated. I think this should preserve whatever's there prior to the addition of the new property.